yaksh/views.py

```diff
@@ -1843,6 +1843,145 @@ def download_assignment_file(request, quiz_id, question_id=None, user_id=None):
+                return my_render_to_response('yaksh/course_detail.html', context,
+                                             context_instance=ci)
+        reader.fieldnames = stripped_fieldnames
+        context['upload_details'] = _read_user_csv(reader, course)
+    return my_render_to_response('yaksh/course_detail.html', context,
+                                 context_instance=ci)
+
+
+def _read_user_csv(reader, course):
+    fields = reader.fieldnames
+    upload_details = ["Upload Summary:"]
+    counter = 0
+    for row in reader:
+        counter += 1
+        (username, email, first_name, last_name, password, roll_no, institute,
+         department, remove) = _get_csv_values(row, fields)
+        if not email or not first_name or not last_name:
+            upload_details.append("{0} -- Missing Values".format(counter))
+            continue
+        users = User.objects.filter(username=username)
+        if users.exists():
+            user = users[0]
+            if remove.strip().lower() == 'true':
+                if _remove_from_course(user, course):
+                    upload_details.append("{0} -- {1} -- User rejected".format(
+                                          counter, user.username))
+                    continue
+            else:
+                if _add_to_course(user, course):
+                    upload_details.append("{0} -- {1} -- User rejected".format(
+                                          counter, user.username))
+            if user not in course.get_enrolled():
+                upload_details.append("{0} -- {1} not added to course".format(
+                    counter, user))
+                continue
+        user_defaults = {'email': email, 'first_name': first_name,
+                         'last_name': last_name}
+        user, created = _create_or_update_user(username, user_defaults)
+        profile_defaults = {'institute': institute, 'roll_number': roll_no,
+                            'department': department, 'is_email_verified': True}
+        _create_or_update_profile(user, profile_defaults)
+        if created:
+            state = "Added"
+            user.set_password(password)
+            user.save()
+            course.students.add(user)
+        else:
+            state = "Updated"
+        upload_details.append("{0} -- {1} -- User {2} Successfully".format(
+                              counter, user.username, state))
+    if counter == 0:
+        upload_details.append("No rows in the CSV file")
+    return upload_details
+
+
+def _get_csv_values(row, fields):
+    roll_no, institute, department = "", "", ""
+    remove = "false"
+    email, first_name, last_name = map(str.strip, [row['email'],
+                                       row['firstname'],
+                                       row['lastname']])
+    if 'password' in fields:
+        password = row['password'].strip()
+        if not password:
+            password = email
+    else:
+        password = email
```

Comment:
I feel password = email is redundant in both if and else. You can do password = email by default and check if the password exists in csv if it exists then assign that value to password variable i.e. password = row['password'].strip()